lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1005,7 +1005,7 @@ bool TypeChecker::typeCheckCatchPattern(CatchStmt *S, DeclContext *DC) {
 
 static bool isDiscardableType(Type type) {
   return (type->is<ErrorType>() ||
-          type->isNever() ||
+          type->isUninhabited() ||
```

Comment:
However, a tuple or struct containing a Never is not uninhabited -- that's a thing to fix at some point, I think.